File: fly-http/src/main/java/com/xia/fly/http/RxHttp.java
package com.xia.fly.http;

import com.xia.fly.http.request.DeleteRequest;
import com.xia.fly.http.request.GetRequest;
import com.xia.fly.http.request.PostRequest;
import com.xia.fly.http.request.PutRequest;

import androidx.annotation.NonNull;

/**
 * @author xia
 * @date 2018/7/29.
 */
public final class RxHttp {

    public static GetRequest get(@NonNull String url) {
        return new GetRequest(url);
    }

    public static PostRequest post(@NonNull String url) {
        return new PostRequest(url);
    }

    public static DeleteRequest delete(@NonNull String url) {
        return new DeleteRequest(url);
    }

    public static PutRequest put(@NonNull String url) {
        return new PutRequest(url);
    }
}

File: fly-http/src/main/java/com/xia/fly/http/RxHttp.kt
package com.xia.fly.http

import com.xia.fly.http.request.DeleteRequest
import com.xia.fly.http.request.GetRequest
import com.xia.fly.http.request.PostRequest
import com.xia.fly.http.request.PutRequest

/**
 * @author xia
 * @date 2018/7/29.
 */
@Suppress("unused")
object RxHttp {

    @JvmStatic
    operator fun get(url: String): GetRequest {
        return GetRequest(url)
    }

    @JvmStatic
    fun post(url: String): PostRequest {
        return PostRequest(url)
    }

    @JvmStatic
    fun delete(url: String): DeleteRequest {
        return DeleteRequest(url)
    }

    @JvmStatic
    fun put(url: String): PutRequest {
        return PutRequest(url)
    }
}
